Docker/test/selenium/specs/quickstatements/quickstatements.js

```diff
@@ -30,28 +41,145 @@ describe( 'QuickStatements Service', function () {
+
+		browser.executeQuickStatement( 'Q1|Sclient_wiki|"Main_Page"' );
 
 		// go look at wikibase
-		browser.url( process.env.MW_SERVER + '/wiki/Item:Q1' );
-		$( '.wikibase-toolbarbutton.wikibase-toolbar-item.wikibase-toolbar-button.wikibase-toolbar-button-add' ).waitForDisplayed();
+		const responseQ1 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q1.json' );
+
+		assert( _.isEmpty( responseQ1.data.entities.Q1.sitelinks ) !== true );
+	} );
+
+	it( 'Should be able to add a property to an item', function () {
+
+		propertyId = browser.call( () => WikibaseApi.getProperty( 'string' ) );
+
+		browser.executeQuickStatement( 'Q1|' + propertyId + '|"Will it blend?"' );
+
+		const responseQ1 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q1.json' );
+		assert( responseQ1.data.entities.Q1.claims[ propertyId ][ 0 ].type === 'statement' );
+
+	} );
+
+	it( 'Should be able to add statement with qualifiers', function () {
+
+		propertyIdItem = browser.call( () => WikibaseApi.getProperty( 'wikibase-item' ) );
+
+		browser.executeQuickStatement( 'Q1|' + propertyIdItem + '|Q1|' + propertyIdItem + '|Q1' );
+
+		const responseQ1 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q1.json' );
+		assert( responseQ1.data.entities.Q1.claims[ propertyId ][ 0 ].type === 'statement' );
+
+	} );
+
+	it( 'Should be able to add a property with "wikibase-item" reference', function () {
+
+		propertyIdItem = browser.call( () => WikibaseApi.getProperty( 'wikibase-item' ) );
+		const propertyNumber = propertyIdItem.replace( 'P', '' );
+		browser.executeQuickStatement( 'Q2|' + propertyIdItem + '|Q2|S' + propertyNumber + '|Q2|S' + propertyNumber + '|Q2' );
+
+		const responseQ2 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q2.json' );
+		assert( ( propertyIdItem in responseQ2.data.entities.Q2.claims ) === true );
+	} );
+
+	it( 'Should be able to add a property with "url" reference', function () {
+
+		propertyURL = browser.call( () => WikibaseApi.getProperty( 'url' ) );
+		const url = '"https://www.wikidata.org"';
+		const propertyNumber = propertyURL.replace( 'P', '' );
+		browser.executeQuickStatement( 'Q2|' + propertyIdItem + '|Q3|S' + propertyNumber + '|' + url );
+
+		const responseQ2 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q2.json' );
+		assert( ( propertyURL in responseQ2.data.entities.Q2.claims ) === true );
+	} );
+
+	it( 'Should be able to add a property with "string" reference', function () {
+
+		const stringValue = '"some string"';
+		const propertyNumber = propertyId.replace( 'P', '' );
+		browser.executeQuickStatement( 'Q2|' + propertyIdItem + '|Q3|S' + propertyNumber + '|' + stringValue );
+
+		const responseQ2 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q2.json' );
+		assert( ( propertyURL in responseQ2.data.entities.Q2.claims ) === true );
```

Comment:
This will always succeed if the above test succeeds (as the data in question will be present).